helper/config/decode.go

```diff
@@ -145,3 +146,26 @@ func uint8ToStringHook(f reflect.Kind, t reflect.Kind, v interface{}) (interface
 
 	return v, nil
 }
+
+func updateString(oldString, newString string) string {
+	if oldString != "" {
+		if newString != "" {
+			return newString
+		}
+		return oldString
+	}
+	return newString
+}
+
+func updateMap(oldMap, newMap map[string]string) map[string]string {
+	if oldMap == nil {
+		oldMap = newMap
```

Comment:
I would return newMap here so that the next scope can be out of the else

Reply:
that's cleaner, yeah